### PR TITLE
Patch SgCluster object via composition function

### PR DESCRIPTION
**Pull request by TheBigLee**
## Summary

* P+T compositions don't support maps on bool values, so we're patching the sgcluster now via comp functions
* There was a bug with updating existing objects on the desired array

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

**Comment by Kidswiss** [comment]
We could probably do the fix for the pointer receiver better, but I didn't really want to re-write half the library...

The whole assertion and casting magic started to break as soon as I introduced the pointer receiver. I guess that was the reason we didn't go with them in the first place...

**Review by glrf — APPROVED** [review]
LGTM and I think the library fix is fine

File: functions/vshn-postgres-func/encrypted_pvc.go [hunk 1]
```diff
@@ -4,22 +4,30 @@ import (
...
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
 )
 
 // AddPvcSecret adds a secret for the encrypted PVC for the  PostgreSQL instance.
 func AddPvcSecret(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	err := runtime.AddToScheme(stackgresv1.SchemeBuilder.SchemeBuilder)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot add scheme builder to scheme", err)
+	}
```
[the remark this answers is not in the document]
Comment (Kidswiss) [reply functions/vshn-postgres-func/encrypted_pvc.go:25]: I think after rebasing to master this should actually already be the case, right?

**Review by zugao — APPROVED** [review]
LGTM

**Pull request merged**